### PR TITLE
Add dockerignore and templates

### PR DESCRIPTION
**Pull request by yanivefraim**
We want to add a `.dockerignore` and `templates` folder, both mandatory for fullstack projects

### CODE REVIEW

File: packages/yoshi-common/src/copy-docker.ts [hunk 1]
```diff
@@ -38,4 +43,13 @@ export default async function (config: Config, cwd = process.cwd()) {
...
     fs.outputFileSync(dockerDestinationPath, transformedContent);
   }
+
+  const dockerignoreDestinationPath = path.resolve(cwd, DOCKER_IGNORE);
+  const projectDockerignoreExists = fs.existsSync(dockerignoreDestinationPath);
+  if (!projectDockerignoreExists) {
+    const dockerIgnore = ['node_modules', 'target'];
+    fs.writeFileSync(dockerignoreDestinationPath, dockerIgnore.join('\n'));
+  }
+  const templatesFolderPath = path.resolve(cwd, TEMPLATES);
```
Comment (ronami) [inline packages/yoshi-common/src/copy-docker.ts:53]: I think we should add a comment explaining why this is needed.

File: packages/yoshi-common/src/copy-docker.ts [hunk 2]
```diff
@@ -38,4 +43,13 @@ export default async function (config: Config, cwd = process.cwd()) {
     );
     fs.outputFileSync(dockerDestinationPath, transformedContent);
   }
+
+  const dockerignoreDestinationPath = path.resolve(cwd, DOCKER_IGNORE);
+  const projectDockerignoreExists = fs.existsSync(dockerignoreDestinationPath);
```
Comment (ronami) [inline packages/yoshi-common/src/copy-docker.ts:48]: small but we can have a function for `writeIfDoesntExists()`

File: packages/yoshi-common/src/copy-docker.ts [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
   if (!projectDockerExists) {
-    const dockerFileName = path.resolve(
+    const dockerFilePath = path.resolve(
       __dirname,
       `./templates/${DOCKER_FILE}`,
     );
 
-    const dockerDestinationPath = path.resolve(cwd, DOCKER_FILE);
-    fs.copyFileSync(dockerFileName, dockerDestinationPath);
+    fs.copyFileSync(dockerFilePath, dockerDestinationPath);
```
Comment (ronami) [inline packages/yoshi-common/src/copy-docker.ts:24]: Docker has params in it (`MAINTAINER {%authorName%}`), I think we need to fill these, similar to how CYA does it. Maybe we can use the same utility.

File: packages/yoshi-common/src/copy-docker.ts [hunk 4]
```diff
@@ -1,22 +1,27 @@
...
   }
-  const projectDockerExists = fs.existsSync(path.resolve(cwd, DOCKER_FILE));
+
+  const dockerDestinationPath = path.resolve(cwd, DOCKER_FILE);
+  const projectDockerExists = fs.existsSync(dockerDestinationPath);
   if (!projectDockerExists) {
-    const dockerFileName = path.resolve(
+    const dockerFilePath = path.resolve(
       __dirname,
       `./templates/${DOCKER_FILE}`,
```
Comment (ronami) [inline packages/yoshi-common/src/copy-docker.ts:21]: small but we can use `path.resolve(__dirname, 'templates', DOCKER_FILE)`

File: packages/yoshi-common/src/copy-docker.ts [hunk 5]
```diff
@@ -38,4 +43,15 @@ export default async function (config: Config, cwd = process.cwd()) {
...
   }
+
+  const dockerignoreDestinationPath = path.resolve(cwd, DOCKER_IGNORE);
+  const projectDockerignoreExists = fs.existsSync(dockerignoreDestinationPath);
+  if (!projectDockerignoreExists) {
+    const dockerIgnore = ['node_modules', 'target'];
+    fs.writeFileSync(dockerignoreDestinationPath, dockerIgnore.join('\n'));
+  }
+  // a templates folder is mandatory in case of a Docker
+  // (docker task will fail in case there's no such folder)
```
Comment (ranyitz) [inline packages/yoshi-common/src/copy-docker.ts:54]: This is a bit sad :)

Thanks for the comment!

Comment (ronami) [reply packages/yoshi-common/src/copy-docker.ts:54]: Yeah, Docker should probably copy it if it exists (not fail otherwise)

**Pull request merged**